aries-mobile-tests/pageobjects/bc_wallet/initialization.py

```diff
@@ -21,41 +21,89 @@ class InitializationPage(BasePage):
     error_details_locator = (AppiumBy.ID, "com.ariesbifold:id/DetailsText")
     error_okay_button_locator = (AppiumBy.ID, "com.ariesbifold:id/Okay")
 
+    def __init__(self, driver):
+        super().__init__(driver)
+        # Instantiate possible Modals and Alerts for this page
+        self.oops_something_went_wrong_modal = OopsSomethingWentWrongModal(driver)
+
     def on_this_page(self):
-        #return super().on_this_page(self.on_this_page_text_locator)
-        #print(self.driver.page_source)
         return self.still_initializing()
 
     def still_initializing(self):
+        # Check for something went wrong modal
+        if self.oops_something_went_wrong_modal.is_displayed():
+            # Get the main error
+            main_error = self.oops_something_went_wrong_modal.get_main_error()
+            # if Timeout error, then raise exception
+            if self.oops_something_went_wrong_modal.is_timeout_error():
+                raise Exception(main_error)
+            else:
+                # Otherwise, Show details, and get the details message and raise exception
+                self.oops_something_went_wrong_modal.select_show_details()
+                detailed_error = self.oops_something_went_wrong_modal.get_detailed_error()
+                raise Exception(f"{main_error}\n{detailed_error}")
```

Comment:
Not sure how python works 100% but if you're not using the message in `except:` below maybe creating it without would be cleaner? `raise Exception` vs `raise Exception(f"{main_error}\n{detailed_error}")`

Reply:
I'm not sure if I'm understanding your question correctly, so I apologize if I'm off track here. I'm actually not in an exception at this point. It is a page object, so a class that represents the BC Wallet initialization page. I'm using appium to check for an error when initializing. Appium won't throw an exception if it happens, I just pick the error element off the app page if it exists, then compose an actual exception and throw it back at the test calling this page object. 

Lel me know if I'm not addressing your concern here.  Greatly appreciate the review.